Library1/Library.fs

```diff
@@ -22,9 +22,389 @@ module FetchData =
+        builder.AddAttribute(13, "class", "alert-link")
+        builder.AddAttribute(14, "href", "https://go.microsoft.com/fwlink/?linkid=870381")
+        builder.AddContent(15, "\n        brief survey\n    ")
+        builder.CloseElement()
+        builder.AddContent(16, "\n    and tell us what you think.\n")
+        builder.CloseElement()
+        builder.AddContent(17, "\n\n")
+
+    // This is to demonstrate how a parent component can supply parameters
+    member val Title : string = Unchecked.defaultof<string> with get, set
+
+namespace BlazorApp1.Pages
+
+open System
+open System.Collections.Generic
+open System.Linq
+open System.Threading.Tasks
+open System.Net.Http
+open Microsoft.AspNetCore.Blazor
+open Microsoft.AspNetCore.Blazor.Components
+open Microsoft.AspNetCore.Blazor.Layouts
+open Microsoft.AspNetCore.Blazor.Routing
+open BlazorApp1
+open BlazorApp1.Shared
+
+// NOTE: _ViewImports.cshtml appear to be required in the BlazorApp.
+[<Microsoft.AspNetCore.Blazor.Layouts.LayoutAttribute(typeof<MainLayout>)>]
+type _ViewImports () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+
+[<Microsoft.AspNetCore.Blazor.Layouts.LayoutAttribute(typeof<MainLayout>)>]
+[<Microsoft.AspNetCore.Blazor.Components.RouteAttribute("/")>]
+type Index () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+        builder.AddContent(0, "\n")
+        builder.OpenElement(1, "h1")
+        builder.AddContent(2, "Hello, world!")
+        builder.CloseElement()
+        builder.AddContent(3, "\n\nWelcome to your new app.\n\n")
+        builder.OpenComponent<SurveyPrompt>(4)
+        builder.AddAttribute(5, "title", "How is Blazor working for you?")
+        builder.CloseComponent()
+        builder.AddContent(6, "\n")
+
+(*
+[<Microsoft.AspNetCore.Blazor.Layouts.LayoutAttribute(typeof<MainLayout>)>]
+[<Microsoft.AspNetCore.Blazor.Components.RouteAttribute("/counter")>]
+type Counter () =
+    inherit Microsoft.AspNetCore.Blazor.Components.BlazorComponent()
+    
+    let mutable currentCount = 0
+
+    override this.BuildRenderTree(builder) =
+        base.BuildRenderTree(builder)
+        builder.OpenElement(0, "h1")
+        builder.AddContent(1, "Counter")
+        builder.CloseElement()
+        builder.AddContent(2, "\n\n")
+        builder.OpenElement(3, "p")
+        builder.AddContent(4, "Current count: ")
+        builder.AddContent(5, currentCount)
+        builder.CloseElement()
+        builder.AddContent(6, "\n\n")
+        builder.OpenElement(7, "button")
+        // TODO: determine how to create the onclick RenderTree.RenderTreeFrame
+        builder.AddAttribute(8, onclick(IncrementCount))
+        // NOTE: this may be an option ...
+        //builder.AddAttribute(8, "onclick", UIEventHandler.CreateDelegate(IncrementCount))

```

Comment:
The `onclick` handler generated a `RenderTree.RenderTreeFrame`. I suppose this is buried somewhere else where I couldn't find it. However, the `UIEventHandler.CreateDelegate` looks promising. @danroth27 could probably point me in the right direction.